web/source/kmwuibutton.ts

```diff
@@ -199,100 +200,107 @@ if(!window['keyman']['ui']['name']) {
       if(arguments.length > 0) kbdName = _name;
       if(kbdId)
       {
         if((kbdName == '') || keymanweb['isCJK']())
-        { 
+        {
           kbdId.className='kmw_disabled';
-        } 
+        }
         else
         {
           let osk = keymanweb.osk;
           kbdId.className = osk && osk['isEnabled']() ? 'kmw_show' : 'kmw_hide';
         }
       }
-    }  
-  
+    }
+
     ui.registerEvents = function() {
       let osk = keymanweb.osk;
       if(!osk) {
         return;
       }
       /**
        * UI Functions called by KeymanWeb or OSK
-       */     
-      osk['addEventListener']('show', function(oskPosition) { 
+       */
+      osk['addEventListener']('show', function(oskPosition) {
         var t=keymanweb['getLastActiveElement']();
         if(t)
-        {  
+        {
           if(!oskPosition['userLocated'])
           {
             oskPosition['x'] = util['getAbsoluteX'](t);
             oskPosition['y'] = util['getAbsoluteY'](t)+t.offsetHeight;
           }
-        } 
-        
+        }
+
         ui._ShowKeyboardButton();
-        return oskPosition; 
-      }); 
+        return oskPosition;
+      });
 
-      /* TODO: why is this still needed??? Does it actually do anything?? */ 
-      osk['addEventListener']('hide', function(hiddenByUser) { 
+      /* TODO: why is this still needed??? Does it actually do anything?? */
+      osk['addEventListener']('hide', function(hiddenByUser) {
         if((arguments.length > 0) && hiddenByUser)
         {
           var _a = document.getElementById('KMW_Keyboard');
           if(_a) _a.className = 'kmw_hide';
-        }    
-      }); 
+        }
+      });
     }
-    
+
     /**
      * Show or hide the OSK (always visible for CJK keyboards)
-     * 
+     *
      * @param       {Object}    _anchor     anchor element (?)
-     * @return      {boolean}   
-     **/   
-    ui._ShowKeymanWebKeyboard = function(_anchor) 
-    { 
+     * @return      {boolean}
+     **/
+    ui._ShowKeymanWebKeyboard = function(_anchor)
+    {
       var kbdId=document.getElementById("KMW_Keyboard");
       let osk = keymanweb.osk;
 
-      if((kbdId.className!='kmw_disabled') && osk && osk['show']) 
+      if((kbdId.className!='kmw_disabled') && osk && osk['show'])
       {
         if(osk['isEnabled']()) osk['hide'](); else osk['show'](true);
       }
       if(window.event) window.event.returnValue=false;
 
-      keymanweb['focusLastActiveElement']();    
-      return false;    
+      keymanweb['focusLastActiveElement']();
+      return false;
     }
 
     /**
      * Initialize Button User Interface
-     **/   
+     **/
     ui['initialize'] = ui.Initialize = function() {
       //Never initialize UI before KMW (parameters will be undefined)
-      if(!keymanweb['initialized'])
-      {
-        window.setTimeout(ui.Initialize,250); return;
+      if(!keymanweb['initialized']) {
+        ui.initTimer = window.setTimeout(ui.Initialize,250);
```

Comment:
I see this has a timeout of 250 whereas the other init has a timeout of 50. I can't see any good reason to keep this at 250. I reckon 50 is better -- polling every 50msec until init won't be very expensive.

Reply:
Just re-read this; it looks like you're thinking of `setInterval`, rather than `setTimeout`.  I could go the `setInterval` route, but then I'd need to ignore your comments below about where to place `clearTimeout` - we should only clear it once we're confirmed to be initializing.

Reply:
Imagine this is called twice before keymanweb.initialized. This would setup two timeouts -- we'd lose the id for the first timeout. (e.g. on the third call, keymanweb.initialized==true. First timeout would not be cancelled, so we'd get a second init call. It may not matter, but it isn't tidy.)

Reply:
Okay, so that's a gap in the code as it sits, but that doesn't exactly answer the "polling" / `setInterval` vs `setTimeout` question.  Unless you were thinking of the 'second timeout' as something akin to a 'polling' call - but that feels like a bit much of an inference to make without clearer confirmation.

Reply:
I'm thinking of `setTimeout` in the circumstance where this init function is called multiple times. Not `setInterval`. If you do this, you'll get two calls to `bar()`:

```
  function bar() {
    console.log('bar');
  }
  function foo() {
    window.setTimeout(bar, 100);
  }
  foo();
  foo();
```

Reply:
> polling every 50msec until init won't be very expensive.

Consider that this currently sets up a timeout every 250msec. So, it's effectively 'polling'.